wgpu-native/src/instance.rs

```diff
@@ -200,6 +215,26 @@ pub extern "C" fn wgpu_instance_get_adapter(
     HUB.adapters.register_local(adapter)
 }
 
+#[cfg(all(feature = "local", feature = "gfx-backend-gl"))]
+pub fn wgpu_gl_create_surface(windowed_context: back::glutin::WindowedContext) -> SurfaceId {
+    let raw = back::Surface::from_window(windowed_context);
+    let surface = SurfaceHandle::new(raw);
+    HUB.surfaces.register_local(surface)
+}
+
+#[cfg(feature = "gfx-backend-gl")]
+#[no_mangle]
+pub extern "C" fn wgpu_gl_surface_get_adapter(
```

Comment:
I wonder if we can still have `Instance` types and (some) entry points on GL but merged with surfaces, i.e. like today we do `type CommandEncoderId = CommandBufferId`, we should consider `type SurfaceId = InstanceId` for GL only. Of course, this should be removed once GL gets proper support for headless instances...

Reply:
I actually tried to do this the first time, but I think I tried incorrectly to keep `SurfaceId` and `InstanceId` separate and was stymied by `Surface` not being clone-able.  I'll see if there is more of the API surface that can be kept common with `type SurfaceId = InstanceId`.